### PR TITLE
wtip-003: airlines platform PoC

### PR DESCRIPTION
**Pull request by ondrejchmelar**
https://github.com/windingtree/wtips/issues/10

### CODE REVIEW

File: src/contracts.js [hunk 1]
```diff
@@ -53,13 +55,23 @@ class Contracts {
...
-   * Returns a representation of <a href="https://github.com/windingtree/wt-contracts/blob/v0.2.0/contracts/WTIndex.sol">WTIndex.sol</a>.
+   * Returns a representation of <a href="https://github.com/windingtree/wt-contracts/blob/v0.3.0/contracts/WTHotelIndex.sol">WTHotelIndex.sol</a>.
    *
    * @param  {string} address
    * @return {web3.eth.Contract} Instance of an Index
    */
-  async getIndexInstance (address) {
-    return this._getInstance('index', WTIndexContractMetadata.abi, address);
+  async getHotelIndexInstance (address) {
+    return this._getInstance('index', WTHotelIndexContractMetadata.abi, address);
```
Comment (JirkaChadima) [inline src/contracts.js]: I'd differentiate between hotel and airline index in the `name` parameter of `_getInstance`. It influences caching and using the same name potentially creates a problem. On the other hand as the cache also uses contract address in it's key, maybe the whole name attribute can be dropped.

File: src/contracts.js [hunk 2]
```diff
@@ -86,8 +108,9 @@ class Contracts {
     if (!this.eventRegistry) {
       this.eventRegistry = Object.assign(
         {},
-        generateEventSignatures(WTIndexContractMetadata.abi),
-        generateEventSignatures(HotelContractMetadata.abi)
+        generateEventSignatures(WTHotelIndexContractMetadata.abi),
+        generateEventSignatures(HotelContractMetadata.abi),
+        generateEventSignatures(AirlineContractMetadata.abi),
```
Comment (JirkaChadima) [inline src/contracts.js]: `WTAirlineIndexContractMetadata` is missing here if I'm not mistaken

File: src/data-model/wt-airline-index.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+ * Ethereum smart contract backed implementation of Winding Tree
+ * index wrapper. It provides methods for working with airline
+ * contracts.
+ */
+class WTAirlineIndex extends AbstractWTIndex implements WTAirlineIndexInterface {
+  constructor (indexAddress: string, web3Utils: Utils, web3Contracts: Contracts) {
+    super();
+    this.address = indexAddress;
+    this.web3Utils = web3Utils;
+    this.web3Contracts = web3Contracts;
```
Comment (JirkaChadima) [inline src/data-model/wt-airline-index.js]: Isn't the assignment duplicate to the constructor of the parent `AbstractWTIndex`?

File: src/data-model/wt-hotel-index.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+ * Ethereum smart contract backed implementation of Winding Tree
+ * index wrapper. It provides methods for working with hotel
+ * contracts.
+ */
+class WTHotelIndex extends AbstractWTIndex implements WTHotelIndexInterface {
+  constructor (indexAddress: string, web3Utils: Utils, web3Contracts: Contracts) {
+    super();
+    this.address = indexAddress;
+    this.web3Utils = web3Utils;
+    this.web3Contracts = web3Contracts;
```
Comment (JirkaChadima) [inline src/data-model/wt-hotel-index.js]: Isn't the assignment duplicate to the constructor of the parent `AbstractWTIndex`?

File: src/index.js [hunk 5]
```diff
@@ -2,9 +2,13 @@
...
 import type { OffChainDataClientOptionsType } from './off-chain-data-client';
-import type { WTIndexInterface, AdaptedTxResultsInterface, OffChainDataAdapterInterface, WalletInterface, KeystoreV3Interface } from './interfaces';
-import DataModel from './data-model';
+import type { AdaptedTxResultsInterface, OffChainDataAdapterInterface, WalletInterface, KeystoreV3Interface } from './interfaces/base-interfaces';
+import type { WTHotelIndexInterface } from './interfaces/hotel-interfaces';
+import type { WTAirlineIndexInterface } from './interfaces/airline-interfaces';
+import { AbstractDataModel, AirlineDataModel, HotelDataModel } from './data-model/index';
 import OffChainDataClient from './off-chain-data-client';
+import { HOTEL_SEGMENT_ID } from '../test/utils/data-hotel-model-definition';
+import { AIRLINE_SEGMENT_ID } from '../test/utils/data-airline-model-definition';
```
Comment (JirkaChadima) [inline src/index.js]: Importing from `test` in `src` seems really sketchy to me. It also totally affects the builds.

File: src/data-model/wt-airline-index.js (hunk 3 above)
Comment (ondrejchmelar) [reply src/data-model/wt-airline-index.js]: The abstract class has no constructor (well, the default one - I didn't want to do some dark magic to forbid instantiation, the name should serve as a warning). I thought it better to be explicit and make it harder to work with the ancestor. Do you think it should have a concrete constructor?

Comment (JirkaChadima) [reply src/data-model/wt-airline-index.js]: Oh, right. My bad, I originally looked at a bad diff.

**Pull request merged**